tests/e2e/gc_rwx_multi_ns_gc.go

```diff
@@ -0,0 +1,293 @@
+			if pv != nil {
+				ginkgo.By("Deleting the PV1")
+				err = client.CoreV1().PersistentVolumes().Delete(ctx, pv.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		//Add a check to validate CnsVolumeMetadata crd
+		verifyCRDInSupervisorWithWait(ctx, f, pvcNameInSV, crdCNSVolumeMetadatas, crdVersion, crdGroup, true)
+
+		ginkgo.By("Deleting the PVC")
+		err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pvclaim = nil
+
+		ginkgo.By("Verifying if volume still exists in the Supervisor Cluster")
+		fcdIDInCNS = getVolumeIDFromSupervisorCluster(pvcNameInSV)
+		gomega.Expect(fcdIDInCNS).NotTo(gomega.BeEmpty())
+
+		// Creating label for PV
+		// PVC will use this label as Selector to find PV
+		staticPVLabels := make(map[string]string)
+
+		ginkgo.By("Creating the PV in guest cluster")
+		pv2 := getPersistentVolumeSpecForRWX(pvcNameInSV, v1.PersistentVolumeReclaimDelete,
+			staticPVLabels, diskSize, "", v1.ReadWriteMany)
+		pv2, err = client.CoreV1().PersistentVolumes().Create(ctx, pv2, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			if pv2 != nil {
+				ginkgo.By("Deleting the PV2")
+				err = client.CoreV1().PersistentVolumes().Delete(ctx, pv2.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
+
+		ginkgo.By("Creating the PVC in guest cluster's new namespace")
+		pvc2 := getPersistentVolumeClaimSpecForRWX(newnamespace, staticPVLabels, pv2.Name, diskSize)
+		pvc2, err = client.CoreV1().PersistentVolumeClaims(pvc2.Namespace).Create(ctx, pvc2, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			ginkgo.By("Deleting the PVC2")
+			err = fpv.DeletePersistentVolumeClaim(client, pvc2.Name, pvc2.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			var svClient clientset.Interface
+			if k8senvsv := GetAndExpectStringEnvVar("SUPERVISOR_CLUSTER_KUBE_CONFIG"); k8senvsv != "" {
+				svClient, err = k8s.CreateKubernetesClientFromConfig(k8senvsv)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+
+			svcNamespace := os.Getenv("SVC_NAMESPACE")
+
+			// Delete PVC in SVC.
+			svcPVC, err := svClient.CoreV1().PersistentVolumeClaims(svcNamespace).Get(ctx, pvcNameInSV, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Delete the PVC in SVC")
+			err = fpv.DeletePersistentVolumeClaim(svClient, svcPVC.Name, svcPVC.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			//Add a check to validate CnsVolumeMetadata crd
+			verifyCRDInSupervisorWithWait(ctx, f, pvcNameInSV, crdCNSVolumeMetadatas, crdVersion, crdGroup, false)
+
+			ginkgo.By("Deleting the PV1")
+			err = client.CoreV1().PersistentVolumes().Delete(ctx, pv.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
delete fileshare after this

Reply:
Under discussion

Reply:
We discussed and concluded that the delete file share is not required as the SV's PVs Reclaim policy is set to Delete - the underlying Fileshare is expected to be deleted on deleting the PV in SV